### PR TITLE
Alter `nullable` constraint when `alterNullable` is set to true

### PR DESCRIPTION
**Pull request by yhaskell**
As discussed in the https://github.com/knex/knex/issues/4401 Issue, I created a PR to make it so nullable alterations only apply when `({ alterNullable: true })` is provided as a param to `.alter()` function (which is a default value, since we do not want to break the previous behaviour).

@elhigu please help further :)

### CODE REVIEW

**Comment by Cpaczek** [comment]
This is also an issue on CockroachDB. See: strapi/strapi#11738

**Comment by kibertoad** [comment]
@OlivierCavadenti this might also look good in 1.0.0

**Comment by intech** [comment]
@OlivierCavadenti before doing anything, check out this post https://github.com/strapi/strapi/issues/11738#issuecomment-1007860155

**Comment by ethan-gallant** [comment]
Can we get this merged? It's causing quite a headache for various projects depending on knex.

**Comment by kibertoad** [comment]
@ethan-gallant tests are missing and there are conflicts. wiuld you like to help?

**Comment by OlivierCavadenti** [comment]
> @OlivierCavadenti before doing anything, check out this post [strapi/strapi#11738 (comment)](https://github.com/strapi/strapi/issues/11738#issuecomment-1007860155)

I can check tomorrow I think. I need to reread the thread to undersant all the problems ^^.

**Review by kibertoad — APPROVED** [review]

**Comment by kibertoad** [comment]
@OlivierCavadenti Do you think we can publish next semver minor now?

**Comment by intech** [comment]
@OlivierCavadenti as it turned out, this is irrelevant, there is a problem with the requests themselves, just a similar error

File: test/integration2/schema/set-nullable.spec.js [hunk 1]
```diff
@@ -74,6 +74,43 @@ describe('Schema', () => {
               id_not_nullable: null,
             });
           });
+
+          it('should throw error if alter a not nullable column with primary key #4401', async function () {
+            if (!isPostgreSQL(knex)) {
```
Comment (kibertoad) [inline test/integration2/schema/set-nullable.spec.js:79]: @OlivierCavadenti Probably can also be run on CockroachDB?

**Pull request merged**